### DISCUSSION TITLE
Ethernet ESP32 support broken with latest IDF44 (fixed)

### DISCUSSION THREAD

**Discussion by Jason2866** [espressif/arduino-esp32#6318] [Q&A] [answered]
compiled Arduino s3 branch with Arduino lib builder.
Somewhere between the relased core 2.0 2 and the actual one is "something"
which brokes ethernet for ESP32 (thinking a IDF change did)

**Comment by VojtechBartoska** [comment]
Hi @Jason2866, thanks for your report.

Can you provide more info?

Board, what exactly doesn't work and so on. **I would suggest to open new issue and fill out issue template.**

Thanks!

**Comment by Jason2866** [comment]
@VojtechBartoska  We are investigating. Boards which do **not** use gpio 17 for Ethernet are working.
If we can provide relevant info we are going to open a issue. 
A log
```
00:00:00.250 Project tasmota - Olimex ESP32-PoE-Iso Version 11.0.0.2(theo)-2_0_2_3(2022-02-21T12:12:39)
I (363) gpio: GPIO[0]| InputEn: 0| OutputEn: 0| OpenDrain: 0| Pullup: 1| Pulldown: 0| Intr:0 
I (364) gpio: GPIO[0]| InputEn: 1| OutputEn: 0| OpenDrain: 0| Pullup: 0| Pulldown: 0| Intr:0 
I (370) gpio: GPIO[17]| InputEn: 0| OutputEn: 0| OpenDrain: 0| Pullup: 1| Pulldown: 0| Intr:0 
I (378) gpio: GPIO[17]| InputEn: 0| OutputEn: 1| OpenDrain: 0| Pullup: 0| Pulldown: 0| Intr:0 
I (386) gpio: GPIO[17]| InputEn: 0| OutputEn: 0| OpenDrain: 0| Pullup: 1| Pulldown: 0| Intr:0 
I (395) gpio: GPIO[17]| InputEn: 0| OutputEn: 0| OpenDrain: 0| Pullup: 0| Pulldown: 0| Intr:0 
E (514) lan87xx: lan87xx_pwrctl(409): power up timeout
E (514) lan87xx: lan87xx_init(491): power control failed
E (514) esp_eth: esp_eth_driver_install(223): init phy failed
00:00:00.413 ETH: Bad PHY type or init error
```

**Comment by SuGlider** [comment]
FYI, I also had problems with GPIO17 and 18 when using them with UART1.
I had to send a PR to use other pins by default. 
The issue happens when the pins are left floating. 
Maybe it has to be pulled up/down externally?

**Comment by Jason2866** [comment]
@SuGlider Thank you, for the hint. Adding a resistor is not a solution. We use different ready made devices which are working fine with there factory firmware (for example https://templates.blakadder.com/ewelink_ZB-GW03.html or https://templates.blakadder.com/sonoff_SPM.html) and replace this firmware with a custom firmware (Tasmota). All fail which uses GPIO17 for clock which is build with actual Arduino version. Core 2.0.2 did work.

**Comment by Jason2866** [comment, marked as answer]
Adding this settings to sdkconfig
```
CONFIG_ETH_RMII_CLK_OUTPUT=y
CONFIG_ETH_RMII_CLK_OUT_GPIO=17
```
solved

**Reply by me-no-dev** [reply]
push it :D https://github.com/espressif/esp32-arduino-lib-builder/blob/esp32-s3-support/configs/defconfig.esp32
nice find!

**Reply by Jason2866** [reply]
@me-no-dev https://github.com/espressif/esp32-arduino-lib-builder/pull/60

**Reply by Jason2866** [reply]
Better fix https://github.com/espressif/arduino-esp32/pull/6340
